src/main/java/com/chikli/hudson/plugin/naginator/NaginatorCause.java

```diff
@@ -9,15 +12,38 @@
  */
 public class NaginatorCause extends Cause {
 
-
     private final String summary;
+    private final Integer sourceBuildNumber;
+    private transient Job project;
 
     public NaginatorCause(AbstractBuild<?, ?> build) {
         this.summary = build.getDisplayName();
+        this.sourceBuildNumber = build.getNumber();
     }
 
     @Override
     public String getShortDescription() {
         return Messages.NaginatorCause_Description(summary);
     }
+
+    @Override
+    public void onAddedTo(AbstractBuild build) {
+        this.project = build.getParent();
+    }
+
+    @Override
+    public void onLoad(@Nonnull AbstractBuild<?,?> build) {
+        this.project = build.getParent();
+    }
+
+    public String getSummary() { return this.summary; }
+
+    public Job getProject() { return this.project; }
+
+    public String getJobUrl() {
+        return this.project.getUrl();
+    }
+
+    public int getSourceBuildNumber() { return this.sourceBuildNumber; }
```

Comment:
This causes NPE for causes created with old versions, as this is compiled as `this.sourceBuildNumber.intValue()`.
Change the return type to `Integer`.